pkg/scheduler/queue/query_component_load.go

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package queue
+
+import (
+	"math"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type QueryComponent string
+
+const (
+	StoreGateway QueryComponent = "store-gateway"
+	Ingester     QueryComponent = "ingester"
+)
+
+var QueryComponents = []QueryComponent{
+	StoreGateway,
+	Ingester,
+}
+
+// cannot import constants from frontend/v2 due to import cycle
+// these are attached to the request's AdditionalQueueDimensions by the frontend.
+const ingesterQueueDimension = "ingester"
+const storeGatewayQueueDimension = "store-gateway"
+const ingesterAndStoreGatewayQueueDimension = "ingester-and-store-gateway"
+
+// QueryComponentNameForRequest parses the expected query component from annotations by the frontend.
+func QueryComponentNameForRequest(req *SchedulerRequest) string {
```

Comment:
I can add a comment for now - ideally now that we moved away from this idea of the queue structure as hyper flexible / extensible structure, I would similarly like to move away from "Additional Queue Dimensions" as it doesn't make sense any more.

This concept of "Expected Query Component" can get a proper name and be put into the `Hints` on the RPC message, rather than continuing this dance of "the expected query component is the first item of the additional queue dimensions array, also there's no other use for this array".